ts/components/LoadingSpinnerOverlay.tsx

```diff
@@ -0,0 +1,78 @@
+import { Button, Text } from "native-base";
+import * as React from "react";
+import { StyleSheet } from "react-native";
+import I18n from "../i18n";
+import variables from "../theme/variables";
+import BoxedRefreshIndicator from "./ui/BoxedRefreshIndicator";
+import { Overlay } from "./ui/Overlay";
+
+const styles = StyleSheet.create({
+  textCaption: {
+    padding: variables.contentPadding
+  },
+  cancelButtonStyle: {
+    alignSelf: "center"
+  }
+});
+
+type Props = Readonly<{
+  isLoading: boolean;
+  loadingCaption?: string;
+  loadingOpacity?: number;
+  onCancel?: any;
```

Comment:
what about declaring it as function?
```suggestion
  onCancel?: () => void;
```